Backend data type definition & Frontend user input data UI

dusixian opened the pull request:
# *Backend data type definition & Frontend user input data UI*

## :recycle: Current situation & Problem
The project requires backend data structures to store manually entered health data and a corresponding frontend UI for users to input these values. This PR introduces data models and a input interface for manual health data entry.


## :gear: Release Notes 
Backend Data Type Definitions:
- Heart Rate (HeartRateEntry) – measured in BPM.
- Temperature (TemperatureEntry) – stored in Celsius or Fahrenheit with unit handling.
- Oxygen Saturation (OxygenSaturationEntry) – percentage validation included.
- Blood Pressure (BloodPressureEntry) – systolic & diastolic values with validation.
- Lab Results (LabEntry) – categorized using LabTestType enum.
- Implemented isValidDate() function to ensure data timestamps are not set in the future.

Frontend UI for Manual Data Entry:

1. DataInputForm View:
    - Displays an input form based on the selected health data type.
    - Includes date & time pickers for accurate timestamping.
    - Supports multiple field types (text fields, number inputs, pickers).
    - Dynamically validates input to ensure correct data entry.
2. AddData View:
    -  Displays 6 data categories for user to choose.  

## :books: Documentation
*Please ensure that you properly document any additions in conformance to [Spezi Documentation Guide](https://github.com/StanfordSpezi/.github/blob/main/DOCUMENTATIONGUIDE.md).*
*You can use this section to describe your solution, but we encourage contributors to document your reasoning and changes using in-line documentation.* 


## :white_check_mark: Testing
Unit Tests Ensured numeric validation for all manual entry fields (e.g., temperature, blood pressure). Confirmed correct data structure initialization with sample values.


## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

PSchmiedmayer commented:
@dusixian As noted in https://github.com/orgs/CS342/discussions/113#discussioncomment-12051489, ensure that you update all dependencies to the latest version in Xcode and committing all the updated project files + Package.resolved file. You need to run File -> Swift Packages -> Update to Latest Package Versions in Xcode. This might fix your current testing errors.

mmervecerit commented:
Hi @dusixian,

I reviewed your PR, here is my feedback:

- Your changes in eventview and scheduleview should be reverted i believe, since those lines were added by Paul to the original template to fix issues, and i added them in my previous PR.
-Also the units of lab tests are important to know, and critical to input/show in UI. (hemoglobin for example, it is g/dL and WBC is usually cells per microliter) This might be good to talk with @viraj28m when he is implementing FHIR records of the lab values and have unit conversions if required.

dusixian commented:
@mmervecerit Hi Merve! Thanks for your advice. I've changed the eventview and scheduleview following your commits. And for the unit of lab variables, I changed the frontend so the current UI shows the unit when user entry the lab values.
